### PR TITLE
USWDS-Site: Added side navigation accessibility test page

### LINKED ISSUE #2857: CC a11y: publish side navigation checklist - create PR
Publish the `usa-side-navigation` accessibility test checklist items found in the [Critical Checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?usp=sharing) (Google Doc :lock:). 

The process for creating a new accessibility test page can be found in the [Creating accessibility checklists - process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit#heading=h.fa3sv1ray42g) (Google Doc :lock:).

### PR DESCRIPTION
**Pull request by RachelCorsino**
# Summary

Added accessibility test page for side navigation component

## Related issue

Closes #2857 

## Preview link
Preview link: [Side Navigation Component Page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-side-nav-checklist/components/side-navigation/)

## Testing and review
Follow these steps:

1. Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
2. Confirm that the main component page links to the accessibility tests page in the side navigation.
3. Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
4. Confirm that the main component page shows the accessibility tests summary.
5. Check that accessibility tests page provides the correct counts for each test status type.
6. Confirm the test checklist summaries and data are accurate.
7. Confirm no visual issues.
8. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

### CODE REVIEW

File: _components/sidenav/accessibility-tests.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+permalink: /components/side-navigation/accessibility-tests/
+layout: accessibility-test
+component:
+  name: side navigation
+title: Side navigation accessibility tests
+category: Components
+lead: Any USWDS [component name] component should pass these manual accessibility tests.
```
Comment (mejiaj) [inline _components/sidenav/accessibility-tests.md:8]:
```suggestion
lead: Any USWDS side navigation component should pass these manual accessibility tests.
```

File: _components/sidenav/sidenav.md [hunk 2]
```diff
@@ -7,19 +7,11 @@ component:
...
-  href: '#side-navigation-preview'
-- text: Code
-  href: '#side-navigation-code'
-- text: Guidance
-  href: '#side-navigation-guidance'
-- text: Package
-  href: '#side-navigation-package'
-- text: Latest updates
-  href: '#changelog'
+  - text: side navigation accessibility tests
```
Comment (mejiaj) [inline _components/sidenav/sidenav.md:13]:
```suggestion
  - text: Side navigation accessibility tests
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
Thanks for creating this @RachelCorsino. 

I've updated the base branch, made some formatting edits, and made two small suggestions.

File: _components/sidenav/accessibility-tests.md (hunk 1 above)
Comment (RachelCorsino) [reply _components/sidenav/accessibility-tests.md]: Resolved in [9c30372](https://github.com/uswds/uswds-site/pull/2864/commits/9c3037216def30ecfc367e0adf4292a569722918)

File: _components/sidenav/sidenav.md (hunk 2 above)
Comment (RachelCorsino) [reply _components/sidenav/sidenav.md]: Resolved in [9c30372](https://github.com/uswds/uswds-site/pull/2864/commits/9c3037216def30ecfc367e0adf4292a569722918)

File: _components/sidenav/sidenav.md [hunk 3]
```diff
@@ -7,19 +7,11 @@ component:
 lead: "Hierarchical, vertical navigation to place at the side of a page."
 permalink: /components/side-navigation/
 redirect_from:
-- /sidenav/
-- /components/sidenav/
+  - /sidenav/
+  - /components/sidenav/
```
Comment (mahoneycm) [inline _components/sidenav/sidenav.md:11]: **polish:** Remove added whitespace

```suggestion
- /sidenav/
- /components/sidenav/
```

File: _data/accessibility-tests/side-navigation.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    test_status: conditional
+    test_type: keyboard
+    version_tested: 3.8.2
+    wcag_criterion: 2.1.2
+  - summary: Focus indicator is clearly visible in side navigation.
+    summary_additional: When you use a keyboard to tab into the side navigation, the links have a visible underline, bold style or other clear indication that links are interactive.
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.2
+    wcag_criterion: 2.4.7
```
Comment (mahoneycm) [inline _data/accessibility-tests/side-navigation.yml:47]: **question (non-blocking):** Was there a reason we changed the Gherkin for focus indicators? This mentions a visible underline and bold style but neither is the default for focus indicators. Most of our other descriptions for `2.4.7` read:

>**Focus indicator is clearly visible.** When you use a keyboard to navigate through the header, there will be a visible outline or other clear indication where the the focus is. 

 @amycole501  @alex-hull @finekatie

**Review by mahoneycm — CHANGES_REQUESTED** [review]
Looks good! One small change and one question for a11y

File: _data/accessibility-tests/side-navigation.yml (hunk 4 above)
Comment (alex-hull) [reply _data/accessibility-tests/side-navigation.yml:47]: Hmm...I'm not sure why that one got changed. I think it should reflect the way others have been written. The previous way potentially helps with some generalizing, versus tying people to those options.

File: _components/sidenav/sidenav.md (hunk 3 above)
Comment (mejiaj) [reply _components/sidenav/sidenav.md:11]: @mahoneycm The indent follows existing conventions.

File: _data/changelogs/component-side-navigation.yml [hunk 5]
```diff
@@ -2,57 +2,62 @@ title: Side navigation
...
-  affectsAccessibility: false
-  affectsAssets: false
-  affectsGuidance: false
-  affectsJavascript: false
-  affectsMarkup: false
-  affectsStyles: true
-  githubPr: 4163
-  githubRepo: uswds
-  versionUswds: 2.11.2
+  - date: NNNN-NN-NN
```
Comment (mejiaj) [inline _data/changelogs/component-side-navigation.yml:5]: **chore:**  Will need to add date before final merge.

File: _data/changelogs/component-side-navigation-accessibility.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+title: Side navigation accessibility tests
+type: component
+items:
+  - date: NNNN-NN-NN
```
Comment (mejiaj) [inline _data/changelogs/component-side-navigation-accessibility.yml:4]: **chore:**  Will need to add date before final merge.

**Review by mejiaj — APPROVED** [review]
Code looks good and follows standards. Re-running CI checks because of a potential timeout.

File: _data/accessibility-tests/side-navigation.yml (hunk 4 above)
Comment (RachelCorsino) [reply _data/accessibility-tests/side-navigation.yml]: Let me know if this change is what you had in mind! [77b383e](https://github.com/uswds/uswds-site/pull/2864/commits/77b383ec6991c2c440096a1d3188ad9887b846b6)

**Comment by RachelCorsino** [comment]
Also updated the test status for `2.4.8`. Resolved in [8ac3f58](https://github.com/uswds/uswds-site/pull/2864/commits/8ac3f5883edd2fd80390cb382f9042d211808c50)

**Review by mahoneycm — APPROVED** [review]
LGTM!

**Review by alex-hull — APPROVED** [review]
Looks good! Approved!

File: _data/changelogs/component-side-navigation-accessibility.yml (hunk 6 above)
Comment (thisisdano) [inline _data/changelogs/component-side-navigation-accessibility.yml:4]:
```suggestion
  - date: 2024-10-16
```

File: _data/changelogs/component-side-navigation.yml (hunk 5 above)
Comment (thisisdano) [inline _data/changelogs/component-side-navigation.yml:5]:
```suggestion
  - date: 2024-10-16
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**